### PR TITLE
[Improvement] Add optional environment variables

### PR DESCRIPTION
**Pull request by izchen**
### What changes were proposed in this pull request?

Added six optional environment variables:

```
# RSS_HOME, RSS home directory (Default: parent directory of the script)
# RSS_CONF_DIR, RSS configuration directory (Default: ${RSS_HOME}/conf)
# HADOOP_CONF_DIR, Hadoop configuration directory (Default: ${HADOOP_HOME}/etc/hadoop)
# RSS_PID_DIR, Where the pid file is stored (Default: ${RSS_HOME})
# RSS_LOG_DIR, Where log files are stored (Default: ${RSS_HOME}/logs)
# RSS_IP, IP address Shuffle Server binds to on this node (Default: first non-loopback ipv4)
```

### Why are the changes needed?

Simplify service deployment

### Does this PR introduce _any_ user-facing change?

Added six optional environment variables.

 With the default environment variables, the behavior is the same as before this PR.

### How was this patch tested?

local test

### CODE REVIEW

**Comment by colinmjj** [comment]
How about unify `COORDINATOR_HOME` and `SHUFFLE_SERVER_HOME` to `RSS_HOME`
All new environment variables can be defined in `rss-env.sh`

**Comment by izchen** [comment]
> How about unify `COORDINATOR_HOME` and `SHUFFLE_SERVER_HOME` to `RSS_HOME`
> All new environment variables can be defined in `rss-env.sh`

I think we can define `RSS_HOME` and `RSS_CONF_DIR` in `rss-env.sh`. I will modify the script later.

File: bin/stop-shuffle-server.sh [hunk 1]
```diff
@@ -20,10 +20,8 @@
 set -o pipefail
 set -o nounset   # exit the script if you try to use an uninitialised variable
 set -o errexit   # exit the script if any statement returns a non-true return value
-set -e
```
Comment (izchen) [inline bin/stop-shuffle-server.sh:23]: unify style

`set -e` same as `set -o errexit`: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html

**Comment by izchen** [comment]
> How about unify `COORDINATOR_HOME` and `SHUFFLE_SERVER_HOME` to `RSS_HOME`
> All new environment variables can be defined in `rss-env.sh`

Done.

Would you mind take a look when you have some time? @colinmjj

**Comment by colinmjj** [comment]
LGTM, +1, thanks for the contribution, @izchen

**Review by jerqi — APPROVED** [review]
LGTM +1, @izchen thanks

**Comment by izchen** [comment]
Thanks for your review, @colinmjj @jerqi !

**Pull request merged**